crates/leon/src/values.rs

```diff
@@ -0,0 +1,37 @@
+use std::{
+    collections::{BTreeMap, HashMap},
+    hash::Hash,
+};
+
+pub trait Values<K, V> {
+    fn get_value(&mut self, _key: K) -> Option<V>;
+}
+
```

Comment:
IMO it's probably a good idea to also `impl<K, V, T> Values<K, V> for &mut T where T: Values<K, V>`

Reply:
ah, yes, that also makes the hash/btree map impls simpler